### PR TITLE
Full Cargo Sbom generation

### PR DESCRIPTION
**Pull request by dzakirin-muhammad**
- 🚚 Added cycloneDx sbom generation #24 
- ‎🔥🔥🔥 Included hotfix for BuildToolsExist #25 
- 📦 Added sample toml file

### CODE REVIEW

**Review by norazlina-md-nor — APPROVED** [review]

File: adapters/cargo/cargo.go [hunk 1]
```diff
@@ -14,27 +14,31 @@ type Cargo struct {
 
 func (cargo *Cargo) Generate(resultInfo structs.ResultInfo) []byte {
 
+	console.Info("Starting SBOM generation for Cargo !! ")
+
 	workingDir := "SBOMWorkingDir/" + resultInfo.Uuid + "/"
 	err := utils.CreateFolder(workingDir)
 	if err != nil {
-		log.Printf("Unable to create %v !! ", workingDir)
+		console.Error("Unable to create working dir folder !! ")
```
Comment (hazimhasnan-edag) [inline adapters/cargo/cargo.go:22]: I thought we standardized using console for command line execution and using normal log library for our program log

File: adapters/cargo/cargo.go [hunk 2]
```diff
@@ -14,27 +14,31 @@ type Cargo struct {
...
+		console.Error(err.Error())
 	}
 
 	err = utils.CreateFolder(workingDir + "src/")
 	err = os.WriteFile(workingDir+"src/main.rs", []byte(" "), 0644)
 	if err != nil {
+		console.Error("Unable to src/main.rs !! ")
```
Comment (soon-yoong) [inline adapters/cargo/cargo.go:29]: minor typo `Unable to create src/main.rs`

File: adapters/cargo/cargo.go (hunk 1 above)
Comment (soon-yoong) [reply adapters/cargo/cargo.go:22]: ya i think we should use `log` as console does not print to log files

Comment (dzakirin-muhammad) [reply adapters/cargo/cargo.go:22]: change to logging only on this specific line or to all @hazimhasnan-edag @soon-yoong  ?

File: adapters/cargo/cargo.go (hunk 2 above)
Comment (dzakirin-muhammad) [reply adapters/cargo/cargo.go:29]: typo fixed 🔧

File: adapters/cargo/cargo.go (hunk 1 above)
Comment (dzakirin-muhammad) [reply adapters/cargo/cargo.go]: changes applied 🔄

**Review by hazimhasnan-edag — APPROVED** [review]

**Pull request merged**